src/player.cpp

```diff
@@ -5855,23 +5857,21 @@ comfort_level player::base_comfort_value( const tripoint &p ) const
             comfort += static_cast<int>( comfort_level::very_comfortable );
         }
     } else if( plantsleep ) {
-        if( vp || furn_at_pos != f_null ) {
-            // Sleep ain't happening in a vehicle or on furniture
+        // Sleep ain't happening in a vehicle or on inappropriate furniture
+        if( vp || !( g->m.has_flag_furn( "ALLOW_ROOT", pos() ) ||  g->m.has_flag_furn( "ALLOW_ROOT_WEAK", pos() ) ) ){
```

Comment:
If the furniture there has the `ALLOW_ROOT_WEAK` flag, but doesn't have the `ALLOW_ROOT` flag, then you won't be able to use it, because the `!has_flag_furn( ALLOW_ROOT )` check will return true and prevent that from being checked.
Also, this needs the not on the third check, because otherwise furnitures with `ALLOW_ROOT_WEAK` will never give the comfort level they are supposed to give on line 5870, because this will be triggered first.
```suggestion 
        if( vp || ( !( g->m.has_flag_furn( "ALLOW_ROOT", pos() ) &&  !g->m.has_flag_furn( "ALLOW_ROOT_WEAK", pos() ) ) ) ){
```
But after all that, you don't need to check the furniture here - they'll be caught in the later checks, on line 5865 and 5869.
```suggestion
        if( vp ){
```

Reply:
This statement goes as this:
if in vehicle OR NOT (furn has flag OR furn has other flag) then can't sleep
then (when we know there is no vehicle and no furniture without flags)
if furniture or terrain has flag can sleep
then if furniture has other flag can sleep (not as good)
and then everything else
This is made this way because otherwise you can sleep on top of furniture which is placed on terrain with proper flag.

Reply:
Okay, my bad, I missed the parenthesis.